tests/CMakeLists.txt

```diff
@@ -193,17 +193,25 @@ foreach(test ${TESTS})
     WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}")
 endforeach()
 
+if(WIN32)
+    set(test_ext ".bat")
+endif()
+
 # MAC tests
 foreach(test
-  hmac-md5
-  hmac-md5-96
-  hmac-sha1
-  hmac-sha1-96
-  hmac-sha2-256
-  hmac-sha2-512
+  read_hmac-md5
+  read_hmac-md5-96
+  read_hmac-sha1
+  read_hmac-sha1-96
+  read_hmac-sha2-256
+  read_hmac-sha2-512
+  read_hmac-sha1-etm
+  read_hmac-sha2-256-etm
+  read_hmac-sha2-512-etm
   )
-  add_test(NAME test_${test} COMMAND "$<TARGET_FILE:test_read>")
-  set_tests_properties(test_${test} PROPERTIES ENVIRONMENT "FIXTURE_TEST_MAC=${test}")
+  add_test(
+      NAME test_${test} COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/test_${test}${test_ext}")
+  set_tests_properties(test_${test} PROPERTIES ENVIRONMENT "builddir=$<TARGET_FILE_DIR:test_read>")
```

Comment:
This would revert 05c7d128c8966f60ec168790432e6e76e91db9db. I suggest replacing it with:
```diff
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index c748e81..28d02c9 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -201,6 +201,9 @@ foreach(test
   hmac-sha1-96
   hmac-sha2-256
   hmac-sha2-512
+  hmac-sha1-etm
+  hmac-sha2-256-etm
+  hmac-sha2-512-etm
   )
   add_test(NAME test_${test} COMMAND "$<TARGET_FILE:test_read>")
   set_tests_properties(test_${test} PROPERTIES ENVIRONMENT "FIXTURE_TEST_MAC=${test}")
```

Reply:
The test binaries include the MAC name in the filename and these end with -etm@openssh.com

It wasn't a problem to just have these filenames when testing on Linux but I wonder if this will cause problems on Windows and this was one of the reasons to have scripts and batch files for each MAC read test.

Perhaps it will be possible to only include the part of the MAC name before @ in the test filename to avoid Windows problems, but I will wait on results from CI.

Reply:
With current master the separate scripts are never used. CMake runs `test_read[.exe]` for each test, passing the parameter via an envvar. Windows can handle script filenames with `@` in them fine, if we get to use those in the future.